airbyte-cdk/bulk/toolkits/load-azure-blob-storage/src/main/kotlin/io/airbyte/cdk/load/file/azureBlobStorage/AzureBlobStreamingUpload.kt

```diff
@@ -94,4 +96,22 @@ class AzureBlobStreamingUpload(
         // Encode the entire fixed-length buffer to Base64
         return Base64.getEncoder().encodeToString(buffer.array())
     }
+    /**
+     * Return a new map containing only valid key/value pairs according to Azure metadata
+     * constraints.
+     */
+    private fun filterInvalidMetadata(metadata: Map<String, String>): Map<String, String> {
+        return metadata.filter { (key, value) -> isValidKey(key) && isValidValue(value) }
+    }
+
+    private fun isValidKey(key: String): Boolean {
+        if (key.isBlank() || key.startsWith(RESERVED_PREFIX)) return false
+        if (invalidCharsRegex.containsMatchIn(key)) return false
+        return key.all { it.code in 32..126 }
```

Comment:
Nit: add comment here about the character range (e.g. it is all ASCII characters, etc)